diff --git a/scripts/gentoo/nova-agent.in b/scripts/gentoo/nova-agent.in
index 0d11ac2..2286f34 100755
--- a/scripts/gentoo/nova-agent.in
+++ b/scripts/gentoo/nova-agent.in
@@ -30,12 +30,26 @@ pidfile="/var/run/nova-agent.pid"
 logfile="/var/log/nova-agent.log"
 loglevel="info"
 
+depend() {
+  before net
+  after xe-daemon
+}
+
 start() {
     ebegin "Starting nova-agent"
     start-stop-daemon --start --exec ${nova_agent} \
         --pidfile ${pidfile} \
         -- -q -p ${pidfile} -o ${logfile} -l ${loglevel} ${agent_config}
-    eend $?
+    RET=$?
+
+    if [ "x${UUID}" != "x$(/usr/bin/xenstore read name)" ]; then
+        /usr/bin/xenstore write data/host/system-init '{"name":"resetnetwork","value":""}'
+        /usr/bin/xenstore watch -n 1 data/guest/system-init >/dev/null
+        [ "$(/usr/bin/xenstore read data/guest/system-init | wc -l)" -gt "0" ] || eend 1
+        echo "UUID=\"$(/usr/bin/xenstore read name)\"" >> /etc/conf.d/nova-agent
+    fi
+    
+    eend $RET
 }
 
 stop() {
