diff --git a/pycallgraph/config.py b/pycallgraph/config.py
index 5911fef..e3492c1 100755
--- a/pycallgraph/config.py
+++ b/pycallgraph/config.py
@@ -34,7 +34,7 @@ class Config(object):
         self.did_init = True
 
         # Update the defaults with anything from kwargs
-        [setattr(self, k, v) for k, v in kwargs.iteritems()]
+        [setattr(self, k, v) for k, v in kwargs.items()]
 
         self.create_parser()
 
diff --git a/pycallgraph/output/output.py b/pycallgraph/output/output.py
index 9660d58..2e921ab 100644
--- a/pycallgraph/output/output.py
+++ b/pycallgraph/output/output.py
@@ -16,7 +16,7 @@ class Output(object):
         self.edge_label_func = self.edge_label
 
         # Update the defaults with anything from kwargs
-        [setattr(self, k, v) for k, v in kwargs.iteritems()]
+        [setattr(self, k, v) for k, v in kwargs.items()]
 
     def set_config(self, config):
         '''
