diff --git a/tests/unit/test_module.py b/tests/unit/test_module.py
index 645ea2d..ffb0a5d 100644
--- a/tests/unit/test_module.py
+++ b/tests/unit/test_module.py
@@ -424,15 +424,6 @@ class PyraxInitTest(unittest.TestCase):
                 region=None)
         pyrax.connect_to_cloud_databases.assert_called_once_with(region=None)
 
-    @patch('pyrax._cs_client.Client', new=fakes.FakeCSClient)
-    def test_connect_to_cloudservers(self):
-        pyrax.cloudservers = None
-        sav = pyrax.connect_to_cloudservers
-        pyrax.connect_to_cloudservers = self.orig_connect_to_cloudservers
-        pyrax.cloudservers = pyrax.connect_to_cloudservers()
-        self.assertIsNotNone(pyrax.cloudservers)
-        pyrax.connect_to_cloudservers = sav
-
     @patch('pyrax.StorageClient', new=fakes.FakeService)
     def test_connect_to_cloudfiles(self):
         pyrax.cloudfiles = None
@@ -468,18 +459,6 @@ class PyraxInitTest(unittest.TestCase):
         pyrax.cloud_databases = pyrax.connect_to_cloud_databases()
         self.assertIsNotNone(pyrax.cloud_databases)
 
-    def test_set_http_debug(self):
-        pyrax.cloudservers = None
-        sav = pyrax.connect_to_cloudservers
-        pyrax.connect_to_cloudservers = self.orig_connect_to_cloudservers
-        pyrax.cloudservers = pyrax.connect_to_cloudservers()
-        pyrax.cloudservers.http_log_debug = False
-        pyrax.set_http_debug(True)
-        self.assertTrue(pyrax.cloudservers.http_log_debug)
-        pyrax.set_http_debug(False)
-        self.assertFalse(pyrax.cloudservers.http_log_debug)
-        pyrax.connect_to_cloudservers = sav
-
     def test_get_encoding(self):
         sav = pyrax.get_setting
         pyrax.get_setting = Mock(return_value=None)
