diff --git a/doc/source/conf.py b/doc/source/conf.py
index f256fd4..30d0259 100644
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -42,7 +42,6 @@ sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__),
 extensions = ['sphinx.ext.autodoc',
               'sphinx.ext.todo',
               'sphinx.ext.coverage',
-              'sphinx.ext.intersphinx',
               'oslosphinx',
               # NOTE(blk-u): Uncomment the [pbr] section in setup.cfg and
               # remove this Sphinx extension when
@@ -233,8 +232,3 @@ latex_documents = [
 
 # If false, no module index is generated.
 #latex_use_modindex = True
-
-intersphinx_mapping = {
-    'python': ('http://docs.python.org/', None),
-    'osloconfig': ('http://docs.openstack.org/developer/oslo.config/', None),
-}
