Index: pidgin-knotify.c
===================================================================
--- pidgin-knotify.c	(revision 18)
+++ pidgin-knotify.c	(working copy)
@@ -27,6 +27,7 @@
 
 #include <glib.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #include "connection.h"
 #include "conversation.h"
@@ -64,16 +65,22 @@
 notify(const gchar *title,
 	   const gchar *body)
 {
-	gchar *command = NULL;
 	int result, timeout;
+  char timeout_str[255];
 
 	timeout = purple_prefs_get_int("/plugins/knotify/notify_timeout");
-	command = g_strdup_printf("kdialog --title '%s' --passivepopup '%s' %d", title, body, timeout);
+  sprintf(timeout_str, "%d", timeout);
+  
+  char * args[] = {
+    "kdialog",
+    "--title",
+    title,
+    "--passivepopup",
+    body,
+    timeout
+    };
 
-	if (command != NULL) {
-			result = system(command);
-			g_free(command);
-	}
+  result = execvp(args[0], args);
 }
 
 /* Signon flood be gone! - thanks to pidgin-libnotify */
