diff --git a/setup.py b/setup.py
index 3b97afb..69bdd91 100755
--- a/setup.py
+++ b/setup.py
@@ -19,6 +19,7 @@
 import ast
 import os
 import re
+from code import open
 from setuptools import setup, find_packages
 
 
@@ -50,7 +51,7 @@ def parse_requirements(path):
     function properly.
     """
     try:
-        requirements = map(str.strip, local_file(path).splitlines())
+        requirements = map(unicode.strip, local_file(path).splitlines())
     except IOError:
         raise RuntimeError("Couldn't find the `requirements.txt' file :(")
 
@@ -70,7 +71,7 @@ def parse_requirements(path):
 
 
 local_file = lambda *f: \
-    open(os.path.join(os.path.dirname(__file__), *f)).read()
+    open(os.path.join(os.path.dirname(__file__), *f), encoding = 'utf-8').read()
 
 
 install_requires, dependency_links = \
