diff -rupN kdeplasma-addons-4.4.5/dataengines/microblog/oauth.cpp kdeplasma-addons-4.4.5.new//dataengines/microblog/oauth.cpp
--- kdeplasma-addons-4.4.5/dataengines/microblog/oauth.cpp	2010-06-25 13:40:10.000000000 -0500
+++ kdeplasma-addons-4.4.5.new//dataengines/microblog/oauth.cpp	2010-07-06 19:04:54.423244177 -0500
@@ -120,7 +120,8 @@ QByteArray createSignature(const QString
 //     kDebug() << "SIG BASE STRING: " << signatureBaseString;
 
     if (!QCA::isSupported("hmac(sha1)")) {
-        qFatal("HMAC(SHA1) is not supported!");
+        kError() << "Hashing algo not supported, update your QCA";
+        return QByteArray();
     }
     // create key for HMAC-SHA1 hashing
     QByteArray key(ConsumerSecret + "&" + tokenSecret);
