diff --git a/Makefile b/Makefile
index 949aee2..de5aca7 100644
--- a/Makefile
+++ b/Makefile
@@ -50,12 +50,12 @@ install: $(PROG)
 	@echo "a location of your choosing manually"
 	@echo "--------------------------------------------------"
 	@echo ""
-	cp -f $(PROG) /usr/sbin/
-	cp -f $(PROG).1 /usr/man/man1/ || cp -f $(PROG).1 /usr/local/man/man1/
+	mkdir -p $(PREFIX)/sbin/ && cp -f $(PROG) $(PREFIX)/sbin/
+	mkdir -p $(PREFIX)/share/man/man1 && cp -f $(PROG).1 $(PREFIX)/share/man/man1/
 
 uninstall:
-	rm -f /usr/sbin/$(PROG)
-	rm -f /usr/man/man1/$(PROG).1 || rm -f /usr/local/man/man1/$(PROG).1
+	rm -f $(PREFIX)/sbin/$(PROG)
+	rm -f $(PREFIX)/share/man/man1/$(PROG).1
 
 clean:
 	rm -f $(PROG)
