--- configure.orig	2008-06-19 22:44:34.000000000 +0300
+++ configure	2010-02-11 18:41:43.000000000 +0200
@@ -15627,7 +15627,7 @@
   if test "$withval" != "no"; then
      if test "$withval" = "yes"; then
 	for python_root in /usr /usr/local /usr/sfw; do
-	  for ver in python2.2 python2.3 python2.4 python2.5; do
+	  for ver in python2.2 python2.3 python2.4 python2.5 python2.6; do
 	     if test -f $python_root/include/${ver}/Python.h; then
 		PYTHON_INCDIR=-I$python_root/include/${ver}
 		if test -d $python_root/lib64/${ver}/config; then
